include/rocksdb/db.h

```diff
@@ -1712,6 +1714,18 @@ class DB {
     return Status::NotSupported("GetStatsHistory() is not implemented.");
   }
 
+  virtual Status DumpCache(const CacheDumpOptions& /*dump_options*/,
+                           const std::shared_ptr<Cache> /*cache*/) {
```

Comment:
> Is there a reason to make these methods off of the DB rather than the Cache class itself? It seems like they more logically belong on the Cache itself.

In the finally version, I prefer to keep it just as part of the library. Application creates the CacheDumper instance and do the job, which is fully independent to either Cache or DB.